### PR TITLE
Don't build upstream dependencies with errorprone

### PR DESCRIPTION
**Pull request by nineinchnick**
## Description

Errorprone is only needed to impacted modules, not all
dependencies.

## Additional context and related issues

https://trinodb.slack.com/archives/CP1MUNEUX/p1691199541970659

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Comment by nineinchnick** [comment]
Testing this in https://github.com/nineinchnick/trino/pull/15

**Comment by nineinchnick** [comment]
This is not effective because properties can't override plugin configuration, per [GIB docs](https://github.com/gitflow-incremental-builder/gitflow-incremental-builder/#configuration):
> Just like for regular plugins, a plugin config property that is set to a fixed value cannot be changed via the respective system property (here: -Dgib.uncommitted=false).

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -188,17 +188,14 @@ jobs:
...
+      - name: Maven Install
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-          $MAVEN clean package ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
       - name: Error Prone Checks
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-          # Run Error Prone on one module with a retry to ensure all runtime dependencies are fetched
-          $MAVEN ${MAVEN_TEST} -T 1C clean verify -DskipTests -P gib,errorprone-compiler -am -pl ':trino-spi'
```
Comment (findepi) [inline .github/workflows/ci.yml:199]: remove this in a seaparate prep commit.
this is invalid, there is no retry here

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -188,17 +188,14 @@ jobs:
       - uses: ./.github/actions/setup
         with:
           cache: restore
-      - name: Maven Package
+      - name: Maven Install
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-          $MAVEN clean package ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
+          $MAVEN clean install ${MAVEN_FAST_INSTALL} ${MAVEN_GIB} -pl '!:trino-docs,!:trino-server,!:trino-server-rpm'
```
Comment (findepi) [inline .github/workflows/ci.yml:194]: intuitively, this looks redundant. error-prone can compile on its own (this is how i run it locally)

please add a code comment

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -188,17 +188,14 @@ jobs:
...
       - name: Error Prone Checks
         run: |
           export MAVEN_OPTS="${MAVEN_INSTALL_OPTS}"
-          # Run Error Prone on one module with a retry to ensure all runtime dependencies are fetched
-          $MAVEN ${MAVEN_TEST} -T 1C clean verify -DskipTests -P gib,errorprone-compiler -am -pl ':trino-spi'
-          # The main Error Prone run
-          $MAVEN ${MAVEN_TEST} -T 1C clean verify -DskipTests -P gib,errorprone-compiler \
+          $MAVEN ${MAVEN_TEST} -T 1C clean verify -DskipTests ${MAVEN_GIB} -Dgib.buildUpstream=never -Dgib.buildDownstream=never -P errorprone-compiler \
```
Comment (findepi) [inline .github/workflows/ci.yml:198]: > Errorprone is only needed to check modules with changes, not all
impacted modules, that is downstream dependencies.

i think that's an assumption that's not necessarily always true. a legit change in affected module can cause error-prone failure in a downstream

**for example**, a field change from int to long could cause NarrowingCompoundAssignment failure in eg base-jdbc

Comment (nineinchnick) [reply .github/workflows/ci.yml]: Done!

**Pull request merged**